File: 프로그래머스/unrated/181880. 1로 만들기/1로 만들기.py
def calc(a):
    cnt = 0
    while a != 1:
        if a%2 == 0:
            a //= 2
        else:
            a -= 1
            a //= 2
        cnt += 1
    return cnt
        

def solution(num_list):
    answer = 0
    for num in num_list:
        c = calc(num)
        answer += c
    return answer

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    public int solution(int[] num_list) {
        int result = 0;
        for (int num : num_list) {
            int cnt = 0;
            while(num > 1) {
                if (num % 2 == 0) {
                    num /= 2;
                } else {
                    num -= 1;
                    num /= 2;
                }
                
                cnt += 1;
            }
            result += cnt;
        }
        return result;
    }
}